pkg/registry/core/componentstatus/validator.go

```diff
@@ -58,14 +64,27 @@ type ServerStatus struct {
 	Err string `json:"err,omitempty"`
 }
 
-func (server *Server) DoServerCheck(prober httpprober.HTTPProber) (probe.Result, string, error) {
+func (server *Server) DoServerCheck() (probe.Result, string, error) {
+	// setup the prober
+	if server.Prober == nil {
```

Comment:
the `server.Prober == nil` check outside the sync.Once call will still cause race issues with the assignment inside the sync.Once call